diff --git a/src/config/path.rs b/src/config/path.rs
index 1bfbb3c..296f386 100644
--- a/src/config/path.rs
+++ b/src/config/path.rs
@@ -34,7 +34,6 @@ pub fn get_profiles_path() -> PathBuf {
 /// E.g. ["/etc/inputplumber/devices.d", "/usr/share/inputplumber/devices"]
 pub fn get_devices_paths() -> Vec<PathBuf> {
     let paths = vec![
-        PathBuf::from("./rootfs/usr/share/inputplumber/devices"),
         PathBuf::from("/etc/inputplumber/devices.d"),
         get_base_path().join("devices"),
     ];
@@ -48,7 +48,6 @@ pub fn get_capability_maps_paths() -> Vec<PathBuf> {
     let paths = vec![
         get_base_path().join("capability_maps"),
         PathBuf::from("/etc/inputplumber/capability_maps.d"),
-        PathBuf::from("./rootfs/usr/share/inputplumber/capability_maps"),
     ];
 
     paths
