diff --git a/usr/lib/steamos/steamos-set-plasma-theme b/usr/lib/steamos/steamos-set-plasma-theme
index 8d441c5..4f6871f 100755
--- a/usr/lib/steamos/steamos-set-plasma-theme
+++ b/usr/lib/steamos/steamos-set-plasma-theme
@@ -1,5 +1,2 @@
 #! /bin/bash
-board_name=$(cat /sys/class/dmi/id/board_name)
-if [[ "$board_name" == "Jupiter" || "$board_name" == "Galileo" ]] then
     kwriteconfig6 --file /etc/xdg/kdeglobals --group KDE --key LookAndFeelPackage "com.valve.vapor.deck.desktop"
-fi
