diff -rupN mesa-mesa-24.0.3.orig/src/mesa/main/draw.c mesa-mesa-24.0.3/src/mesa/main/draw.c
--- mesa-mesa-24.0.3.orig/src/mesa/main/draw.c	2024-03-27 00:12:28.967259871 +0000
+++ mesa-mesa-24.0.3/src/mesa/main/draw.c	2024-03-27 00:13:11.869845179 +0000
@@ -475,7 +475,7 @@ static GLenum
 validate_draw_arrays(struct gl_context *ctx,
                      GLenum mode, GLsizei count, GLsizei numInstances)
 {
-   if (count < 0 || numInstances < 0)
+   if (count <= 0 || numInstances <= 0)
       return GL_INVALID_VALUE;
 
    GLenum error = _mesa_valid_prim_mode(ctx, mode);
