diff -rupN linux-6.12.10.orig/sound/pci/hda/patch_realtek.c linux-6.12.10/sound/pci/hda/patch_realtek.c
--- linux-6.12.10.orig/sound/pci/hda/patch_realtek.c	2025-01-23 14:44:15.362339577 +0000
+++ linux-6.12.10/sound/pci/hda/patch_realtek.c	2025-01-23 18:40:31.202914643 +0000
@@ -7793,6 +7793,21 @@ enum {
 	ALC233_FIXUP_MEDION_MTL_SPK,
 };
 
+/* Fix up the audio pinning on the Ayn MiniPro, as it
+ * shares a product and vendor id with other devices. */
+static void alc269_fixup_match_via_dmi(struct hda_codec *codec,
+				       const struct hda_fixup *fix, int action)
+{
+	int alc269_fix_id;
+	const char *board_name = dmi_get_system_info(DMI_BOARD_NAME);
+	if (dmi_name_in_vendors("ayn") && strcmp(board_name, "Loki MiniPro")) {
+		alc269_fix_id = ALC269VB_FIXUP_AYANEO_SPKR_PIN_FIX;
+	} else {
+		alc269_fix_id = ALC269_FIXUP_AYANEO_VOLUME;
+	}
+	__snd_hda_apply_fixup(codec, alc269_fix_id, action, 0);
+}
+
 /* A special fixup for Lenovo C940 and Yoga Duet 7;
  * both have the very same PCI SSID, and we need to apply different fixups
  * depending on the codec ID
@@ -9053,13 +9068,13 @@ static const struct hda_fixup alc269_fix
 			{ 0x1b, 0x90170152 } /* use as internal speaker (back) */
 		}
 	},
-	[ALC269_FIXUP_HEADSET_AYA_2] = {
+	[ALC269_FIXUP_AYANEO_VOLUME] = {
 		.type = HDA_FIXUP_FUNC,
 		.v.func = alc269_fixup_headphone_volume,
 	},
-	[ALC269_FIXUP_HEADSET_AYA_GEEK] = {
+	[ALC269_FIXUP_DMI_MATCH] = {
 		.type = HDA_FIXUP_FUNC,
-		.v.func = alc269_fixup_headphone_volume,
+		.v.func = alc269_fixup_match_via_dmi,
 	},
 	[ALC299_FIXUP_PREDATOR_SPK] = {
 		.type = HDA_FIXUP_PINS,
@@ -11026,8 +11041,7 @@ static const struct hda_quirk alc269_fix
 	SND_PCI_QUIRK(0xf111, 0x0006, "Framework Laptop", ALC295_FIXUP_FRAMEWORK_LAPTOP_MIC_NO_PRESENCE),
 	SND_PCI_QUIRK(0xf111, 0x0009, "Framework Laptop", ALC295_FIXUP_FRAMEWORK_LAPTOP_MIC_NO_PRESENCE),
 	SND_PCI_QUIRK(0xf111, 0x000c, "Framework Laptop", ALC295_FIXUP_FRAMEWORK_LAPTOP_MIC_NO_PRESENCE),
-	SND_PCI_QUIRK(0x1f66, 0x0101, "GEEK", ALC269_FIXUP_HEADSET_AYA_GEEK),
-	SND_PCI_QUIRK(0x1f66, 0x0101, "AYANEO 2", ALC269_FIXUP_HEADSET_AYA_2),
+	SND_PCI_QUIRK(0x1f66, 0x0101, "Multiple Vendors", ALC269_FIXUP_DMI_MATCH),
 	SND_PCI_QUIRK(0x1f66, 0x0103, "AYANEO AIR 1S", ALC269VB_FIXUP_AYANEO_SPKR_PIN_FIX),
 
 #if 0
