diff -rupN linux-6.12.orig/drivers/hwmon/oxp-sensors.c linux-6.12/drivers/hwmon/oxp-sensors.c
--- linux-6.12.orig/drivers/hwmon/oxp-sensors.c	2024-11-17 22:15:08.000000000 +0000
+++ linux-6.12/drivers/hwmon/oxp-sensors.c	2024-11-18 12:34:25.438029683 +0000
@@ -45,12 +45,14 @@ static bool unlock_global_acpi_lock(void
 enum oxp_board {
 	aok_zoe_a1 = 1,
 	aya_neo_2,
+	aya_neo_2s,
 	aya_neo_air,
 	aya_neo_air_1s,
 	aya_neo_air_plus_mendo,
 	aya_neo_air_pro,
 	aya_neo_flip,
 	aya_neo_geek,
+	aya_neo_geek_1s,
 	aya_neo_kun,
 	orange_pi_neo,
 	oxp_2,
@@ -111,6 +113,13 @@ static const struct dmi_system_id dmi_ta
 		},
 		.driver_data = (void *)aya_neo_2,
 	},
+        {
+                .matches = {
+                        DMI_MATCH(DMI_BOARD_VENDOR, "AYANEO"),
+                        DMI_EXACT_MATCH(DMI_BOARD_NAME, "AYANEO 2S"),
+                },
+                .driver_data = (void *)aya_neo_2s,
+        },
 	{
 		.matches = {
 			DMI_MATCH(DMI_BOARD_VENDOR, "AYANEO"),
@@ -128,6 +137,13 @@ static const struct dmi_system_id dmi_ta
 	{
 		.matches = {
 			DMI_MATCH(DMI_BOARD_VENDOR, "AYANEO"),
+			DMI_EXACT_MATCH(DMI_BOARD_NAME, "AIR 1S"),
+		},
+		.driver_data = (void *)aya_neo_air_1s,
+	},
+	{
+		.matches = {
+			DMI_MATCH(DMI_BOARD_VENDOR, "AYANEO"),
 			DMI_EXACT_MATCH(DMI_BOARD_NAME, "AB05-Mendocino"),
 		},
 		.driver_data = (void *)aya_neo_air_plus_mendo,
@@ -169,6 +185,20 @@ static const struct dmi_system_id dmi_ta
 	},
 	{
 		.matches = {
+			DMI_MATCH(DMI_BOARD_VENDOR, "AYANEO"),
+			DMI_EXACT_MATCH(DMI_BOARD_NAME, "GEEK 1S"),
+		},
+		.driver_data = (void *)aya_neo_geek_1s,
+	},
+        {
+		.matches = {
+			DMI_MATCH(DMI_BOARD_VENDOR, "AYANEO"),
+			DMI_MATCH(DMI_BOARD_NAME, "FLIP"),
+		},
+		.driver_data = (void *)aya_neo_flip,
+	},
+	{
+		.matches = {
 			DMI_MATCH(DMI_BOARD_VENDOR, "ONE-NETBOOK"),
 			DMI_EXACT_MATCH(DMI_BOARD_NAME, "ONE XPLAYER"),
 		},
@@ -463,12 +493,14 @@ static int oxp_platform_read(struct devi
 				return read_from_ec(OXP_2_SENSOR_FAN_REG, 2, val);
 			case aok_zoe_a1:
 			case aya_neo_2:
+			case aya_neo_2s:
 			case aya_neo_air:
 			case aya_neo_air_1s:
 			case aya_neo_air_plus_mendo:
 			case aya_neo_air_pro:
 			case aya_neo_flip:
 			case aya_neo_geek:
+			case aya_neo_geek_1s:
 			case aya_neo_kun:
 			case oxp_fly:
 			case oxp_mini_amd:
@@ -588,12 +620,14 @@ static int oxp_platform_write(struct dev
 				val = (val * 184) / 255;
 				return write_to_ec(OXP_SENSOR_PWM_REG, val);
 			case aya_neo_2:
+			case aya_neo_2s:
 			case aya_neo_air:
 			case aya_neo_air_1s:
 			case aya_neo_air_plus_mendo:
 			case aya_neo_air_pro:
 			case aya_neo_flip:
 			case aya_neo_geek:
+			case aya_neo_geek_1s:
 			case aya_neo_kun:
 			case oxp_mini_amd:
 			case oxp_mini_amd_a07:
