From 1c5b0dfc807fac142f004c581817a94bbaca8a48 Mon Sep 17 00:00:00 2001
From: fewtarius <88717793+fewtarius@users.noreply.github.com>
Date: Tue, 28 May 2024 18:54:32 +0200
Subject: [PATCH 40/41] Add Atari VCS quirk

---
 drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c b/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c
index 99dbd2341120..e9ce71fced4b 100644
--- a/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c
@@ -1170,6 +1170,8 @@ static const struct amdgpu_gfxoff_quirk amdgpu_gfxoff_quirk_list[] = {
 	{ 0x1002, 0x15dd, 0x103c, 0x83e7, 0xd3 },
 	/* GFXOFF is unstable on C6 parts with a VBIOS 113-RAVEN-114 */
 	{ 0x1002, 0x15dd, 0x1002, 0x15dd, 0xc6 },
+	/* GFXOFF is unstable on revision 91 (Atari VCS) with 113-RAVEN2-117 */
+	{ 0x1002, 0x15D8, 0x1002, 0x15D8, 0x91 },
 	/* Apple MacBook Pro (15-inch, 2019) Radeon Pro Vega 20 4 GB */
 	{ 0x1002, 0x69af, 0x106b, 0x019a, 0xc0 },
 	{ 0, 0, 0, 0, 0 },
diff -rupN linux-6.10.5.orig/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c linux-6.10.5/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
--- linux-6.10.5.orig/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c	2024-08-14 13:34:38.000000000 +0000
+++ linux-6.10.5/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c	2024-08-28 15:53:20.432207108 +0000
@@ -1883,6 +1883,7 @@ static int amdgpu_device_init_apu_flags(
 		break;
 	case CHIP_RENOIR:
 		if ((adev->pdev->device == 0x1636) ||
+		    (adev->pdev->device == 0x1638) ||
 		    (adev->pdev->device == 0x164c))
 			adev->apu_flags |= AMD_APU_IS_RENOIR;
 		else
--
2.43.2
