diff -rupN linux-6.12.1.orig/drivers/nvme/host/pci.c linux-6.12.1/drivers/nvme/host/pci.c
--- linux-6.12.1.orig/drivers/nvme/host/pci.c	2024-11-22 14:30:26.000000000 +0000
+++ linux-6.12.1/drivers/nvme/host/pci.c	2024-12-04 01:53:27.164321884 +0000
@@ -3487,9 +3487,10 @@ static const struct pci_device_id nvme_i
 				NVME_QUIRK_IGNORE_DEV_SUBNQN, },
 	{ PCI_DEVICE(0x1cc1, 0x33f8),   /* ADATA IM2P33F8ABR1 1 TB */
 		.driver_data = NVME_QUIRK_BOGUS_NID, },
-	{ PCI_DEVICE(0x10ec, 0x5762),   /* ADATA SX6000LNP */
+	{ PCI_DEVICE(0x10ec, 0x5762),   /* ADATA SX6000LNP, Teamgroup MP34, Realtek RTS5762*/
 		.driver_data = NVME_QUIRK_IGNORE_DEV_SUBNQN |
-				NVME_QUIRK_BOGUS_NID, },
+				NVME_QUIRK_BOGUS_NID, 
+				NVME_QUIRK_BROKEN_MSI,},
 	{ PCI_DEVICE(0x10ec, 0x5763),  /* ADATA SX6000PNP */
 		.driver_data = NVME_QUIRK_BOGUS_NID, },
 	{ PCI_DEVICE(0x1cc1, 0x8201),   /* ADATA SX8200PNP 512GB */
