From c16838eeade4789341c3a8fc26beaa08e40a535a Mon Sep 17 00:00:00 2001
From: fewtarius <fewtarius@jelos.org>
Date: Fri, 24 May 2024 17:38:14 +0000
Subject: [PATCH] Add Ayaneo and Ayn platform drivers to kernel.

---
diff -rupN linux-6.9.1.orig/drivers/Kconfig linux-6.9.1/drivers/Kconfig
--- linux-6.9.1.orig/drivers/Kconfig	2024-05-21 20:38:58.279988325 -0400
+++ linux-6.9.1/drivers/Kconfig	2024-05-21 20:40:55.336654955 -0400
@@ -245,4 +245,8 @@ source "drivers/cdx/Kconfig"
 
 source "drivers/dpll/Kconfig"
 
+source "drivers/ayaneo-platform/Kconfig"
+
+source "drivers/ayn-platform/Kconfig"
+
 endmenu
diff -rupN linux-6.9.1.orig/drivers/Makefile linux-6.9.1/drivers/Makefile
--- linux-6.9.1.orig/drivers/Makefile	2024-05-21 20:38:58.279988325 -0400
+++ linux-6.9.1/drivers/Makefile	2024-05-21 20:54:06.723321500 -0400
@@ -198,3 +198,6 @@ obj-$(CONFIG_CDX_BUS)		+= cdx/
 obj-$(CONFIG_DPLL)		+= dpll/
 
 obj-$(CONFIG_S390)		+= s390/
+
+obj-$(CONFIG_AYANEO_PLATFORM)	+= ayaneo-platform/
+obj-$(CONFIG_AYN_PLATFORM)	+= ayn-platform/
