diff -rupN linux-6.9.2.orig/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c linux-6.9.2/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c
--- linux-6.9.2.orig/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c	2024-05-28 09:44:01.917077876 +0000
+++ linux-6.9.2/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c	2024-05-28 10:01:04.504799814 +0000
@@ -1170,6 +1170,8 @@ static const struct amdgpu_gfxoff_quirk
 	{ 0x1002, 0x15dd, 0x103c, 0x83e7, 0xd3 },
 	/* GFXOFF is unstable on C6 parts with a VBIOS 113-RAVEN-114 */
 	{ 0x1002, 0x15dd, 0x1002, 0x15dd, 0xc6 },
+	/* GFXOFF is unstable on 91 (Atari VCS) with 113-RAVEN2-117 */
+	{ 0x1002, 0x15D8, 0x1002, 0x15D8, 0x91 },
 	/* Apple MacBook Pro (15-inch, 2019) Radeon Pro Vega 20 4 GB */
 	{ 0x1002, 0x69af, 0x106b, 0x019a, 0xc0 },
 	{ 0, 0, 0, 0, 0 },
